var/spack/repos/builtin/packages/albany/package.py

```diff
@@ -0,0 +1,128 @@
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Albany(CMakePackage):
+    """Albany is an implicit, unstructured grid, finite element code for the
+       solution and analysis of multiphysics problems.  The Albany repository
+       on the GitHub site contains hundreds of regression tests and examples
+       that demonstrate the code's capabilities on a wide variety of problems
+       including fluid mechanics, solid mechanics (elasticity and plasticity),
+       ice-sheet flow, quantum device modeling, and many other applications."""
+
+    # Add a proper url for your package's homepage here.
+    homepage = "http://gahansen.github.io/Albany"
+    url      = "https://github.com/gahansen/Albany/tarball/master"
+
+    maintainers = ['gahanse']
+
+    # Add proper versions and checksums here.
+    version('master', git='https://github.com/gahansen/Albany.git', branch='master')
+
+    variant('install',          default=True,
+            description='Enable Install')
+    variant('lcm',          default=True,
+            description='Enable LCM')
+    variant('aeras',          default=False,
+            description='Enable AERAS')
+    variant('qcad',          default=False,
+            description='Enable QCAD')
+    variant('hydride',          default=False,
+            description='Enable HYDRIDE')
+    variant('lcm_spec',          default=False,
+            description='Enable LCM_SPECULATIVE')
+    variant('lame',          default=False,
+            description='Enable LAME')
+    variant('edebug',          default=False,
+            description='Enable DEBUGGING')
+    variant('fpe',          default=False,
+            description='Enable CHECK_FPE')
+    variant('scorec',          default=False,
+            description='Enable SCOREC')
+    variant('felix',          default=False,
+            description='Enable FELIX')
+    variant('mor',          default=False,
+            description='Enable MOR')
+    variant('ci',          default=False,
+            description='Enable ALBANY_CI')
+    variant('ascr',          default=False,
+            description='Enable ALBANY_ASCR')
+    variant('perf',          default=False,
```

Comment:
does this run some tests? If so, it's not a variant:

```
spack install -h
--run-tests           run package tests during installation (same as --test=all)
```

Reply:
No - sorry for the confusion. Albany optionally links to a Configuration Interface (CI) that is a GUI that helps users setup the Albany input and execution spec. I could add a better description than "Enable ALBANY_CI" at the expense of making that string longer. Are there limitations on the length of that string (either technical or project policies?)

Reply:
There are no limitations on the length of the description. However, if the line goes over 79 characters, flake8 will complain. As long as you keep the entire `variant(...)` on a single line, flake8 will not check that line.